shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/view/mapper/ProjectSelectionListViewStateMapper.kt

```diff
@@ -0,0 +1,95 @@
+                        mapProjectListItem(
+                            projectWithProgress = it,
+                            trackId = state.track.id,
+                            bestRatedProjectId = bestRatedProjectId,
+                            fastestToCompleteProjectId = fastestToCompleteProjectId,
+                            level = state.track.projectsByLevel.getProjectLevel(it.project.id)
+                        )
+                    },
+                    recommendedProjects = state.recommendedProjects.map {
+                        mapProjectListItem(
+                            projectWithProgress = it,
+                            trackId = state.track.id,
+                            bestRatedProjectId = bestRatedProjectId,
+                            fastestToCompleteProjectId = fastestToCompleteProjectId,
+                            level = state.track.projectsByLevel.getProjectLevel(it.project.id)
+                        )
+                    },
+                    projectsByLevel = state.projectsByLevel.mapValues { (level, projects) ->
+                        projects.map { project ->
+                            mapProjectListItem(
+                                projectWithProgress = project,
+                                trackId = state.track.id,
+                                bestRatedProjectId = bestRatedProjectId,
+                                fastestToCompleteProjectId = fastestToCompleteProjectId,
+                                level = level
+                            )
+                        }
+                    },
+                    isRefreshing = state.isRefreshing
+                )
+            }
+            ProjectSelectionListFeature.ContentState.Error -> ProjectSelectionListFeature.ViewState.Error
+        }
+
+    private fun mapProjectListItem(
+        projectWithProgress: ProjectWithProgress,
+        trackId: Long,
+        bestRatedProjectId: Long?,
+        fastestToCompleteProjectId: Long?,
+        level: ProjectLevel?
+    ): ProjectSelectionListFeature.ProjectListItem =
+        with(projectWithProgress) {
+            ProjectSelectionListFeature.ProjectListItem(
+                id = project.id,
+                title = project.title,
+                averageRating = progress.averageRating(),
+                level = level,
+                timeToComplete = getTimeToComplete(progress.secondsToComplete),
+                isGraduated = project.isGraduated(trackId),
+                isBestRated = project.id == bestRatedProjectId,
+                isIdeRequired = project.isIdeRequired,
+                isFastestToComplete = project.id == fastestToCompleteProjectId
+            )
+        }
+
+    private fun getTimeToComplete(secondsToComplete: Double): String {
+        val hours = floor(secondsToComplete / 3600).toInt()
+
+        return resourceProvider.getQuantityString(SharedResources.plurals.hours, hours, hours)
+    }
```

Comment:
Check out this comment https://github.com/hyperskill/mobile-app/pull/475#discussion_r1191342321